File: leetcode/editor/en/[169]Majority Element.java
169
        Majority Element
        2023-01-14 00:04:30

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        return nums[nums.length / 2];
    }
}
//leetcode submit region end(Prohibit modification and deletion)
/*
class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        return nums[nums.length / 2];
    }
}
 */

File: leetcode/editor/en/[169]Majority Element.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    const bar = Math.floor(nums.length / 2);
    let map = new Map();

    for (const num of nums) {
        map.set(num, (map.get(num) || 0) + 1);
    }

    for (const [key, value] of map) {
        if (value > bar) {
            return key;
        }
    }

    return -1;
};
//leetcode submit region end(Prohibit modification and deletion)
